packages/@aws-cdk/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -0,0 +1,325 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as logs from '@aws-cdk/aws-logs';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct, IResource, PhysicalName, Resource } from '@aws-cdk/core';
+import { CfnFlowLog } from './ec2.generated';
+import { ISubnet, IVpc } from './vpc';
+
+/**
+ * A FlowLog
+ *
+ * @experimental
+ */
+export interface IFlowLog extends IResource {
+  /**
+   * The Id of the VPC Flow Log
+   *
+   * @attribute
+   */
+  readonly flowLogId: string;
+}
+
+/**
+ * The type of VPC traffic to log
+ *
+ * @experimental
+ */
+export enum FlowLogTrafficType {
+  /**
+   * Only log accepts
+   */
+  ACCEPT = 'ACCEPT',
+
+  /**
+   * Log all requests
+   */
+  ALL = 'ALL',
+
+  /**
+   * Only log rejects
+   */
+  REJECT = 'REJECT'
+}
+
+/**
+ * The type of resource to create the flow log for
+ *
+ * @experimental
+ */
+export abstract class FlowLogResourceType {
+  /**
+   * The subnet to attach the Flow Log to
+   */
+  public static fromSubnet(subnet: ISubnet): FlowLogResourceType {
+    return {
+      resourceType: 'Subnet',
+      resourceId: subnet.subnetId
+    };
+  }
+
+  /**
+   * The VPC to attach the Flow Log to
+   */
+  public static fromVpc(vpc: IVpc): FlowLogResourceType {
+    return {
+      resourceType: 'VPC',
+      resourceId: vpc.vpcId
+    };
+  }
+
+  /**
+   * The Network Interface to attach the Flow Log to
+   */
+  public static fromNetworkInterfaceId(id: string): FlowLogResourceType {
+    return {
+      resourceType: 'NetworkInterface',
+      resourceId: id
+    };
+  }
+
+  /**
+   * The type of resource to attach a flow log to.
+   */
+  public abstract resourceType: string;
+
+  /**
+   * The Id of the resource that the flow log should be attached to.
+   */
+  public abstract resourceId: string;
+}
+
+/**
+ * The destination type for the flow log
+ *
+ * @experimental
+ */
+export abstract class FlowLogDestination {
+  /**
+   * Use CloudWatch logs as the destination
+   */
+  public static toCloudWatchLogs(iamRole?: iam.IRole): FlowLogDestination {
```

Comment:
I would have expected `toCloudWatchLogs()` to take a LogGroup as a parameter.

Reply:
If you wanted to do that but couldn't figure out a good way to make a default log group if the user hadn't supplied one, the way we usually do that is by turning `toDestination(): Config` into something like `bind(scope: Construct, flowLog: FlowLog): Config` and calling it like 

```ts
const config = destination.bind(this, this);
```

`bind()` then has the ability to create the default log group under `scope`.

Reply:
@rix0rrr I refactored both `toCloudWatchLogs()` and `toS3()` based on this comment. I moved the creation of default resources to `bind()`. Let me know what you think.